substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -558,7 +615,7 @@ impl ConfigDef {
 			helper::take_first_item_pallet_attr(&mut item.attrs)?;
 		let disable_system_supertrait_check = attr.is_some();
 
-		if !has_frame_system_supertrait && !disable_system_supertrait_check {
+		if is_frame_system && !disable_system_supertrait_check {
```

Comment:
```suggestion
		if !is_frame_system && !disable_system_supertrait_check {
```

Reply:
hmm, if pallet does NOT have a system supertrait -> it is a system pallet - this was the previous assumption.

Reply:
> hmm, if pallet does NOT have a system supertrait -> it is a system pallet - this was the previous assumption.

My suggestion is wrong, but this isn't true some pallet don't have an explicit system supertrait and are not system balance.
For example they can bound pallet balance config and have the disable supertrait check, and it will compile good. I have seen it, not in this codebase.

Maybe we should keep the code as it was before: if there is no system supertrait and if there is no disabling of the check then return the error.

Reply:
do you also mean removing the `is_frame_system` pallet attribute?

Reply:
No the `is_frame_system` attribute is good.

This code `if !has_frame_system_supertrait && !disable_system_supertrait_check { .. }` is to show a good error if they forgot to bound the supertrait `frame_system::Config`. Do you confirm that the error shows up if somebody forgot to bound `frame_system::Config` as supertrait?

Reply:
I see your point, I restored the check for system supertrait